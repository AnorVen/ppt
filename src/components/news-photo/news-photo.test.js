/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, act, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { NewsPhoto } from '.';

const props = {
	onDeleteFile: jest.fn(),
	input: {},
	onSetPhotoChange: jest.fn(),
};

describe('NewsPhoto', () => {
	it('should  render NewsPhoto', () => {
		render(<NewsPhoto {...props} />);
		expect(screen.queryByTestId('news-image-block')).toBeInTheDocument();
		expect(screen.queryByTestId('news-no-image')).toBeInTheDocument();
		expect(screen.queryByTestId('news-image-input')).toBeInTheDocument();
		expect(screen.queryByTestId('news-image')).not.toBeInTheDocument();
		expect(screen.queryByTestId('news__image--remove')).not.toBeInTheDocument();
	});
	it('should  render NewsPhoto with photo', () => {
		render(
			<NewsPhoto
				{...props}
				input={{
					value: 'value',
				}}
			/>,
		);
		expect(screen.queryByTestId('news-image-block')).toBeInTheDocument();
		expect(screen.queryByTestId('news-image')).toBeInTheDocument();
		expect(screen.queryByTestId('news-no-image')).not.toBeInTheDocument();
		expect(screen.queryByTestId('news-image-input')).toBeInTheDocument();
		expect(screen.queryByTestId('news__image--remove')).toBeInTheDocument();
	});
	it('should call onDeleteFile', async () => {
		render(
			<NewsPhoto
				{...props}
				input={{
					value: 'value',
				}}
			/>,
		);

		act(() => {
			userEvent.click(screen.queryByTestId('news__image--remove'));
		});
		await waitFor(() => {
			expect(props.onDeleteFile).toBeCalled();
		});
	});

	it('should call onSetPhotoChange', async () => {
		const file = new File(['test'], 'test.png', { type: 'image/png' });
		render(<NewsPhoto {...props} />);
		act(() => {
			userEvent.upload(document.querySelector('#input__file'), file);
		});
		await waitFor(() => {
			expect(props.onSetPhotoChange).toBeCalled();
		});
	});

	test('snapshot test NewsPhoto', async () => {
		const component = renderer.create(<NewsPhoto {...props} />).toJSON();
		expect(component).toMatchSnapshot();
	});
	test('snapshot test NewsPhoto with photo', async () => {
		const component = renderer
			.create(
				<NewsPhoto
					{...props}
					input={{
						value: 'value',
					}}
				/>,
			)
			.toJSON();

		expect(component).toMatchSnapshot();
	});
});
