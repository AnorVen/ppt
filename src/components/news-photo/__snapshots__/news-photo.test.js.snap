// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsPhoto snapshot test NewsPhoto 1`] = `
<div
  className="news"
  data-testid="news-image-block"
>
  <i
    aria-hidden="true"
    className="image outline massive icon"
    data-testid="news-no-image"
    onClick={[Function]}
  />
  <div
    className="ui input"
    data-testid="news-image-input"
  >
    <input
      id="input__file"
      onChange={[Function]}
      type="file"
    />
  </div>
  <div
    className="news__icon-wrapper"
  >
    <i
      aria-hidden="true"
      className="pencil alternate icon news__icon"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`NewsPhoto snapshot test NewsPhoto with photo 1`] = `
<div
  className="news"
  data-testid="news-image-block"
>
  <img
    alt="news"
    className="news__image"
    data-testid="news-image"
    src="value"
  />
  <div
    className="ui input"
    data-testid="news-image-input"
  >
    <input
      id="input__file"
      onChange={[Function]}
      type="file"
    />
  </div>
  <span
    className="news__image--remove"
    data-testid="news__image--remove"
    onClick={[MockFunction]}
  />
  <div
    className="news__icon-wrapper"
  >
    <i
      aria-hidden="true"
      className="pencil alternate icon news__icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
